### PR TITLE
feat(authz): Handle jwts as entity chains in decision requests

### PR DESCRIPTION
**Pull request by elizabethhealy**
Resolves https://github.com/opentdf/platform/issues/749
- add endpoint GetDecisionsByToken
- remove jwt entity type
- add support for uuid entity type
- add configuration for different jwt entity selectors

### CODE REVIEW

File: service/kas/access/accessPdp.go
Comment (elizabethhealy) [inline service/kas/access/accessPdp.go:27]: can we remove dissems checks since were just using the token?

Comment (strantalis) [reply service/kas/access/accessPdp.go:27]: @biscoe916 Dissems is deprecated in the spec last time I looked. Can we move forward with removing it? 

If we still need to support Dissems this probably isn't the right place for the check anymore. 

cc: @jrschumacher @ttschampel

File: examples/cmd/encrypt.go
Comment (elizabethhealy) [inline examples/cmd/encrypt.go:49]: fixture provisioning is broken so this doesn't work in ci https://github.com/opentdf/platform/actions/runs/9057319882/job/24881138541?pr=759

File: service/authorization/authorization.proto
Comment (pflynn-virtru) [inline service/authorization/authorization.proto:14]: Can we add `TokenCustom`?   like other Custom, `EntityCustom`

**Review by pflynn-virtru — DISMISSED** [review]

File: service/authorization/authorization.proto
Comment (elizabethhealy) [reply service/authorization/authorization.proto:14]: sure! how would that differ from the string jwt? also what would that look like in terms of selector extraction? we would have to assume the token represented by TokenCustom is a map[string]interface{}

File: service/kas/access/accessPdp.go
Comment (damorris25) [reply service/kas/access/accessPdp.go:27]: we should not drop dissems as eventually we will need it, or some equivalent - however, it could be moved somewhere else

**Comment by elizabethhealy** [comment]
https://github.com/opentdf/platform/pull/790 -- follow up with detailed e2e testing

File: service/authorization/authorization.go
Comment (imdominicreed) [inline service/authorization/authorization.go:101]: NIT: clientSecert has a typo

also i think you want clientSecret and clientId to be lowercase since the config is forced lowercase.

File: service/authorization/authorization.go
Comment (strantalis) [inline service/authorization/authorization.go:13]: Can you switch any usage of `golang-jwt` to the `lestrrat-go`.

File: opentdf-dev.yaml
Comment (strantalis) [inline opentdf-dev.yaml:41]: @elizabethhealy 
Is this saying if `client_id` is missing then try to get `preferred_username`? And is the second example saying if `some_claim` is present and equal to `something` then try to get the selector?

File: service/authorization/authorization.proto
Comment (jakedoublev) [inline service/authorization/authorization.proto:191]: Is this id ephemeral and not long-lasting? Is it possible to rely on an index instead of an on-the-fly id if its purpose is to track decision responses by the `repeated ResourceAttribute` within the token decision request?

File: service/authorization/authorization.proto
Comment (jakedoublev) [inline service/authorization/authorization.proto:169]: I've been wondering if we need scopes on GetEntitlements. Is a scoped GetEntitlements call different than a GetDecision request with a known set of attributes and a set of entities?

Comment (jakedoublev) [reply service/authorization/authorization.proto:169]: This comment should be captured in an issue instead of PR comments. Closing out.

File: service/kas/access/accessPdp.go
Comment (strantalis) [reply service/kas/access/accessPdp.go:27]: @elizabethhealy We discussed this today and you can remove the dissems check from kas. 

I created a place holder issue to track re-implementing it with the GetDecision call on the future.

https://github.com/opentdf/platform/issues/812

File: service/authorization/authorization.go
Comment (elizabethhealy) [reply service/authorization/authorization.go:101]: thanks good catch! i changed the case in the config but forgot the change it here

File: service/authorization/authorization.proto
Comment (elizabethhealy) [reply service/authorization/authorization.proto:191]: probably, i can break this off into a separate issue since we do the same thing for entities when getting multiple of them
also this was just a quick fix, the field was supposed to be there and is referenced in the docs but must have gotten removed

Comment (elizabethhealy) [reply service/authorization/authorization.proto:191]: https://github.com/opentdf/platform/issues/814

File: service/authorization/authorization.go
Comment (elizabethhealy) [reply service/authorization/authorization.go]: switched

File: service/entityresolution/entity_resolution.proto
Comment (jakedoublev) [inline service/entityresolution/entity_resolution.proto:116]: This is probably a Nit, but it would likely keep the Casbin route authz policy cleaner and easier to read if we call the RPCs `CreateEntityChainFromJwt` or `BuildEntityChainFromJwt` instead of `Get...` since this is a POST http verb and not a GET?

Comment (jakedoublev) [reply service/entityresolution/entity_resolution.proto:116]: I don't think this is a blocker regardless because this route is internal to the platform (for now) so it can be changed later.

**Review by jakedoublev — DISMISSED** [review]
I'm not a Codeowner so it both needs and should get a further review, but this LGTM per the ADR outcome.

File: service/entityresolution/entity_resolution.proto
Comment (elizabethhealy) [reply service/entityresolution/entity_resolution.proto:116]: i agree i like CreateEntityChainFromJwt better, ill update it

**Review by strantalis — APPROVED** [review]

**Comment by elizabethhealy** [comment]
TODO: followup with additional integration testing for token exchange flows

File: service/authorization/authorization.go
Comment (dmihalcik-virtru) [inline service/authorization/authorization.go:71]: Should this be `clientSecret` or `clientSecert`?

**Pull request merged**
